### ISSUE TITLE
Add Prediction Table Artifacts to Weights & Biases Logger for each Eval Period

### ISSUE THREAD

**Issue by Peter-Devine** [axolotl-ai-cloud/axolotl#741]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [X] I searched previous [Ideas in Discussions](https://github.com/OpenAccess-AI-Collective/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [X] I searched previous [Issues](https://github.com/OpenAccess-AI-Collective/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

Adding to the functionality of https://github.com/OpenAccess-AI-Collective/axolotl/issues/490, I'd like to log the predictions at each eval period.

Currently, when you log the predictions to a table with Weights and Bias, the name is constant:

https://github.com/OpenAccess-AI-Collective/axolotl/blob/a045db02146751548fec57a5d3f31382ce4e5959/src/axolotl/utils/callbacks.py#L509

This means that on the Weights and Bias dashboard, the predictions are overwritten at each evaluation.

Ideally, I'd like to see how the predictions change over the course of training, and so be able to see the predictions at each evaluation period.

### ✔️ Solution

This could be solved as simply as changing the table name to include the number of global steps at the current point in training. I.e.:

```python
wandb.run.log({f"{name} - Predictions vs Ground Truth - Step {state.global_step}": table})  # type: ignore[attr-defined] 
```

Would you like me to write a pull request for this? Or would this functionality (logging at each step) be better if it was optional?

Thanks for the awesome repo!

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this feature has not been requested yet.
- [X] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by winglian** [comment]
@Peter-Devine yes! this would be great! I was wondering about how to style through various eval steps on the table. If you could submit a PR that would be much appreciated!

**Comment by Peter-Devine** [comment]
@winglian I looked into this and it seems like the data is already saved on Weights and Bias' end, just in a slightly unintuitive location on their dashboard. 

If you go from your experiment page and navigate to:

`Artifacts` > `run-[some hash]-EvalPredictionsvsGroundTruth:v0` > `Files` > `Eval - Predictions vs Ground Truth.table.json`

Then you can find the tables output for each eval period.

While the navigation is a bit clunky (maybe any WandB staff reading this could improve table vizualisation?) I find this a better solution to the one I proposed above. My solution would create a new table every time which might lead to excessive scrolling to find each table. Overall, I think there is a workable solution already available with WandB so I'd be happy to leave it as it is. "If it ain't broke..."

Any worries or are you happy to close this?

References: 
[How WandB saves the data](https://docs.wandb.ai/guides/track/log/log-tables#log-a-table-to-a-run)
[How you can view the data with WandB](https://docs.wandb.ai/guides/tables/visualize-tables#how-to-view-two-tables)

**Comment by MB7979** [comment]
When I first used the evaluation table it was possible to step through each evaluation with a scroller. Unfortunately this functionality seems to have disappeared. I'm not sure why. So I've just been manually checking the json artifacts but it's not as convenient. At any rate, thank you to the original creator of the evaluation table PR, it is super useful.

**Comment by mhenrichsen** [comment]
Closing as stale for now. Seems like a good idea though.
